pytorch_lightning/core/datamodule.py

```diff
@@ -366,3 +362,59 @@ def get_init_arguments_and_types(cls) -> List[Tuple[str, Tuple, Any]]:
             name_type_default.append((arg, arg_types, arg_default))
 
         return name_type_default
+
+    @classmethod
+    def from_datasets(
+        cls,
+        train_dataset: Optional[Union[Dataset, List[Dataset]], Dict[str, Dataset]] = None,
+        val_dataset: Optional[Union[Dataset, List[Dataset]]] = None,
+        test_dataset: Optional[Union[Dataset, List[Dataset]]] = None,
+        batch_size: int = 1,
+        num_workers: int = 0,
+    ):
+        r"""
+        Create an instance from torch.utils.data.Dataset.
+
+        Args:
+            train_dataset: (optional) Dataset to be used for train_dataloader()
+            val_dataset: (optional) Dataset or list of Dataset to be used for val_dataloader()
+            test_dataset: (optional) Dataset or list of Dataset to be used for test_dataloader()
+            batch_size: Batch size to use for each dataloader. Default is 1.
+            num_workers: Number of subprocesses to use for data loading. 0 means that the data will be loaded in the main process.
+                number of CPUs available.
+
+        """
+        def dataloader(ds, shuffle=False):
```

Comment:
Can we not use local function definitions but instead make this a static function at the class level? We have experienced pickling errors in the past, so I want to prevent them no matter how unlikely they are.

Reply:
Agree, a `create_dataloader` staticmethod would be better.